octodns/provider/base.py

```diff
@@ -31,12 +35,42 @@ def __init__(self, existing, desired, changes):
             change_counts[change.__class__.__name__] += 1
         self.change_counts = change_counts
 
+        if self.existing:
+            self.log.debug('__init__: Creates=%s, Updates=%s, Deletes=%s'
+                           'Existing=%s',
+                           self.change_counts['Create'],
+                           self.change_counts['Update'],
+                           self.change_counts['Delete'],
+                           len(self.existing.records))
+        else:
+            self.log.debug('__init__: Creates=%s, Updates=%s, Deletes=%s',
+                           self.change_counts['Create'],
+                           self.change_counts['Update'],
+                           self.change_counts['Delete'])
```

Comment:
To avoid having to duplicate the full log line/code you could do something like.

```python
try:
    existing_n = len(self.existing.records)
except AttributeError:
    existing_n = 0

self.log.debug('__init__: Creates=%d, Updates=%d, Deletes=%d, Existing=%d',
                        self.change_counts['Create'], self.change_counts['Update'],
                        self.change_counts['Delete'], existing_n)
```